### PR TITLE
UISkeleton & useImageSize hook updates

### PR DESCRIPTION
**Pull request by vetkamak**

### CODE REVIEW

**Review by SAnatoliiS — APPROVED** [review]

File: stories/chats/src/BubbleMedia/hooks.tsx [hunk 1]
```diff
@@ -162,9 +164,19 @@ export const useImageSize = (data: string | null) => {
...
+        if (originalSize && originalSizeRef.current) {
+            const newImageSize = getImageSize(
+                originalSizeRef.current?.width,
+                originalSizeRef.current?.height,
+                maxImageSize.width,
+                maxImageSize.height,
+            );
+            setImageSize(newImageSize);
+            return;
+        }
```
Comment (sertony) [inline stories/chats/src/BubbleMedia/hooks.tsx]: Looks a little like a double-code from above. Possibly we are to move `originalSize` as a dependency there.... BUT I don't see where the `originalSize` value is really used here..

Comment (vetkamak) [reply stories/chats/src/BubbleMedia/hooks.tsx]: `originalSize` value is used as the initial for originalSizeRef, so when we already know the image's dimensions we won't need to use Image.getSize() to get them

**Review by sertony — APPROVED** [review]

File: stories/chats/src/BubbleMedia/hooks.tsx [hunk 2]
```diff
@@ -140,9 +140,11 @@ export const useImages = (
     }, [data, preview, imageRef, imageSize, onErrorCallback, onLoadCallback]);
 };
 
-export const useImageSize = (data: string | null) => {
+export const useImageSize = (data: string | null, originalSize?: ImageSize) => {
     const [imageSize, setImageSize] = React.useState<ImageSize | null>(null);
-    const originalSizeRef = React.useRef<ImageSize | null>(null);
+    const originalSizeRef = React.useRef<ImageSize | null>(
+        originalSize === undefined ? null : originalSize,
```
Comment (savelichalex) [inline stories/chats/src/BubbleMedia/hooks.tsx]:
```suggestion
        originalSize ?? null
```

File: kit/layout/src/Skeleton/Skeleton.tsx [hunk 3]
```diff
@@ -190,13 +195,18 @@ export function UISkeleton({
 
     const { isVisible, crossDissolveProgress } = useCrossDissolve(visible);
 
+    const containerStyles = [styles.container, styleProp];
+
     return (
-        <UIBackgroundView style={[styles.container, styleProp]} onLayout={onLayout}>
+        <View
+            style={persistStyles ? containerStyles : isVisible ? containerStyles : null}
```
Comment (savelichalex) [inline kit/layout/src/Skeleton/Skeleton.tsx]: Let's extract it to `useMemo`, nested ternaries are hard to reason about :)

**Review by sertony — APPROVED** [review]

**Review by sertony — APPROVED** [review]

File: kit/layout/src/Skeleton/Skeleton.tsx [hunk 4]
```diff
@@ -190,13 +195,21 @@ export function UISkeleton({
 
     const { isVisible, crossDissolveProgress } = useCrossDissolve(visible);
 
+    const containerStyles = React.useMemo(() => {
+        const containerStyles = [styles.container, styleProp];
+        if (persistStyles) {
+            return containerStyles;
+        }
+        return isVisible ? containerStyles : null;
+    }, [isVisible, persistStyles]);
```
Comment (sertony) [inline kit/layout/src/Skeleton/Skeleton.tsx]: @savelichalex have just pointed out that this is a breaking change in API!

I'd better agree with Aleksey that the SkeletonView should remain to be a View without any style related logic.

**Review by sertony — CHANGES_REQUESTED** [review]

**Review by sertony — APPROVED** [review]

**Review by savelichalex — APPROVED** [review]

**Pull request merged**